### PR TITLE
(PC-6699) unavailable hour booking offer

### PR DESCRIPTION
**Pull request by emmajmn**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-6699

## Checklist

I have:

- [x] Made sure the title of my PR follows the convention `[PC-XXX] <summary>`.
- [x] Made sure my feature is working on the relevant real / virtual devices.
- [x] Written **unit tests** for my feature.
- [ ] Added new reusable components to the AppComponents page and communicate to the team on slack
- [x] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** for any added TODO/FIXME \
       (for tech tasks, give the best context about the TODO resolution: what/who/when).

## Screenshots

| \*iOS - [Phone name] | \*Android - [Phone name] | Little - [Phone name] | Maquette |
| -------------------- | :----------------------: | :--------------------: | ----------------------: |
| image | image | image | image |

## Deploy hard

If native code (ios/android) was modified, **after** the PR is merged, on the master branch, upgrade the **app version** (+1 patch):

- if you want an hard deployment of the testing environment, use `yarn trigger:testing:deploy`

### CODE REVIEW

File: src/features/bookOffer/atoms/ChoiceBloc.tsx [hunk 1]
```diff
@@ -1,22 +1,46 @@
...
+  const buttonHeight = getSpacing(20)
+  return (Math.atan2(buttonHeight, buttonWidth + 2) * 180) / Math.PI
+}
+
+const renderStrikeLine = (buttonWidth: number) => <StrikeLine buttonWidth={buttonWidth} />
+
+const getBorderColor = (selected: boolean, disabled?: boolean) => {
+  if (selected) return ColorsEnum.PRIMARY
+  if (disabled) return ColorsEnum.GREY_MEDIUM
+  else return ColorsEnum.GREY_DARK
```
Comment (antoinewg) [inline src/features/bookOffer/atoms/ChoiceBloc.tsx:26]:
```suggestion
 return ColorsEnum.GREY_DARK
```

File: src/features/bookOffer/atoms/ChoiceBloc.tsx [hunk 2]
```diff
@@ -1,22 +1,46 @@
...
+const getStrikeLineAngle = (buttonWidth: number) => {
+  const buttonHeight = getSpacing(20)
+  return (Math.atan2(buttonHeight, buttonWidth + 2) * 180) / Math.PI
+}
+
+const renderStrikeLine = (buttonWidth: number) => <StrikeLine buttonWidth={buttonWidth} />
+
+const getBorderColor = (selected: boolean, disabled?: boolean) => {
+  if (selected) return ColorsEnum.PRIMARY
+  if (disabled) return ColorsEnum.GREY_MEDIUM
```
Comment (antoinewg) [inline src/features/bookOffer/atoms/ChoiceBloc.tsx:25]: si disabled=false?

File: src/features/bookOffer/atoms/HourChoice.tsx [hunk 3]
```diff
@@ -11,19 +13,33 @@ interface Props {
...
-  const textColor = selected ? ColorsEnum.WHITE : ColorsEnum.BLACK
+export const HourChoice: React.FC<Props> = ({
+  hour,
+  price,
+  selected,
+  onPress,
+  testID,
+  isBookable,
+}) => {
+  const renderTextColor = (selected: boolean, disabled: boolean) => {
```
Comment (antoinewg) [inline src/features/bookOffer/atoms/HourChoice.tsx:27]: en dehors du composant dès que tu peux

Comment (antoinewg) [reply src/features/bookOffer/atoms/HourChoice.tsx:27]: renderTextColor => getTextColor

File: src/features/bookOffer/atoms/ChoiceBloc.tsx [hunk 4]
```diff
@@ -1,22 +1,46 @@
...
+const buttonWidth =
+  (Dimensions.get('window').width - 2 * getSpacing(4) - CHOICE_BLOCS_BY_LINE * getSpacing(2)) /
+  CHOICE_BLOCS_BY_LINE
+
+const getStrikeLineAngle = (buttonWidth: number) => {
+  const buttonHeight = getSpacing(20)
+  return (Math.atan2(buttonHeight, buttonWidth + 2) * 180) / Math.PI
+}
+
+const renderStrikeLine = (buttonWidth: number) => <StrikeLine buttonWidth={buttonWidth} />
```
Comment (blixit) [inline src/features/bookOffer/atoms/ChoiceBloc.tsx:21]: ensuite je te propose d'utiliser un JSX plutôt qu'une fonction qui retourne un JSX

```ts
const renderedStrikeLine = <StrikeLine buttonWidth={buttonWidth} />
```

File: src/features/bookOffer/atoms/HourChoice.tsx (hunk 3 above)
Comment (emmajmn) [reply src/features/bookOffer/atoms/HourChoice.tsx:27]: ok pris en compte

File: src/features/bookOffer/atoms/ChoiceBloc.tsx [hunk 5]
```diff
@@ -25,6 +49,7 @@ export const ChoiceBloc: React.FC<Props> = ({ selected, onPress, testID, childre
           <Spacer.Row numberOfSpaces={5} />
         )}
         {children}
+        {disabled && renderStrikeLine(buttonWidth)}
```
Comment (blixit) [inline src/features/bookOffer/atoms/ChoiceBloc.tsx:52]: et ici:

```ts
{disabled && renderedStrikeLine}
```

File: src/features/bookOffer/atoms/ChoiceBloc.tsx (hunk 1 above)
Comment (emmajmn) [reply src/features/bookOffer/atoms/ChoiceBloc.tsx:26]: modifié

File: src/features/bookOffer/components/BookHourChoice.tsx [hunk 6]
```diff
@@ -50,18 +50,28 @@ export const BookHourChoice: React.FC = () => {
...
-                key={stock.id}
-                price={formatToFrenchDecimal(stock.price).replace(' ', '')}
-                hour={formatHour(stock.beginningDatetime).replace(':', 'h')}
-                selected={stock.id === bookingState.stockId}
-                onPress={() => selectStock(stock.id)}
-                testID={`HourChoice${stock.id}`}
-              />
+          {filteredStocks
+            .filter(
+              (stock) => stock.beginningDatetime !== undefined && stock.beginningDatetime !== null
```
Comment (blixit) [inline src/features/bookOffer/components/BookHourChoice.tsx:55]: ca fait beaucoup de filtres et de sort pour le coup, useMemo ?

File: src/features/bookOffer/atoms/ChoiceBloc.tsx (hunk 4 above)
Comment (emmajmn) [reply src/features/bookOffer/atoms/ChoiceBloc.tsx:21]: 👍🏻

**Review by blixit — APPROVED** [review]

**Review by antoinewg — APPROVED** [review]

**Pull request merged**